src/lib.rs

```diff
@@ -154,48 +162,69 @@ impl BarcodeScanner {
 }
 
 /// Map a scanned key to a character
-fn key_to_str(key: evdev::Key) -> Option<char> {
-	match key {
-		evdev::Key::KEY_A => Some('A'),
-		evdev::Key::KEY_B => Some('B'),
-		evdev::Key::KEY_C => Some('C'),
-		evdev::Key::KEY_D => Some('D'),
-		evdev::Key::KEY_E => Some('E'),
-		evdev::Key::KEY_F => Some('F'),
-		evdev::Key::KEY_G => Some('G'),
-		evdev::Key::KEY_H => Some('H'),
-		evdev::Key::KEY_I => Some('I'),
-		evdev::Key::KEY_J => Some('J'),
-		evdev::Key::KEY_K => Some('K'),
-		evdev::Key::KEY_L => Some('L'),
-		evdev::Key::KEY_M => Some('M'),
-		evdev::Key::KEY_N => Some('N'),
-		evdev::Key::KEY_O => Some('O'),
-		evdev::Key::KEY_P => Some('P'),
-		evdev::Key::KEY_Q => Some('Q'),
-		evdev::Key::KEY_R => Some('R'),
-		evdev::Key::KEY_S => Some('S'),
-		evdev::Key::KEY_T => Some('T'),
-		evdev::Key::KEY_U => Some('U'),
-		evdev::Key::KEY_V => Some('V'),
-		evdev::Key::KEY_W => Some('W'),
-		evdev::Key::KEY_X => Some('X'),
-		evdev::Key::KEY_Y => Some('Y'),
-		evdev::Key::KEY_Z => Some('Z'),
-		evdev::Key::KEY_0 => Some('0'),
-		evdev::Key::KEY_1 => Some('1'),
-		evdev::Key::KEY_2 => Some('2'),
-		evdev::Key::KEY_3 => Some('3'),
-		evdev::Key::KEY_4 => Some('4'),
-		evdev::Key::KEY_5 => Some('5'),
-		evdev::Key::KEY_6 => Some('6'),
-		evdev::Key::KEY_7 => Some('7'),
-		evdev::Key::KEY_8 => Some('8'),
-		evdev::Key::KEY_9 => Some('9'),
-		evdev::Key::KEY_ENTER => Some('\n'),
-		evdev::Key::KEY_KPENTER => Some('\n'),
-		_ => None,
-	}
+fn key_to_str(key: evdev::Key, shift_pressed: bool) -> Option<char> {
+    let char = match key {
+        // Digits
+        evdev::Key::KEY_1 => ['1', '!'],
+        evdev::Key::KEY_2 => ['2', '@'],
+        evdev::Key::KEY_3 => ['3', '#'],
+        evdev::Key::KEY_4 => ['4', '$'],
+        evdev::Key::KEY_5 => ['5', '%'],
+        evdev::Key::KEY_6 => ['6', '^'],
+        evdev::Key::KEY_7 => ['7', '&'],
+        evdev::Key::KEY_8 => ['8', '*'],
+        evdev::Key::KEY_9 => ['9', '('],
+        evdev::Key::KEY_0 => ['0', ')'],
```

Comment:
I'm worried that the second symbol is keyboard layout specific. However, I don't know if barcode scanners actually honor keyboard layouts or if they just assume a US layout anyway.

I suppose it would be up to the device driver. But keyboard layouts are handled by Xorg / Wayland / ..., not by evdev. So would it even be possible for a barcode scanner to honor keyboard layouts on Linux?

Reply:
My scanner seems to just put in the key codes for the US ASCII layout, so if I use anything other than that I get incorrect results for certain characters. I am not sure how other scanners might handle that though.

I would say to leave it like that for now. If someone manages to come across a scanner that behaves differently, then the code could be overhauled to take into account keyboard layouts.

Reply:
Probably the scanner doesn't have much of a choice. I just hope there aren't any international barcode scanners that use a local keyboard layout :pray: